### DISCUSSION TITLE
Typst equivalent of latex-environment extension?

### DISCUSSION THREAD

**Discussion by baptiste** [quarto-dev/quarto-cli#9508] [Q&A]
### Description

It would be great to have something like [latex-environment](https://github.com/quarto-ext/latex-environment) for typst, where the `[content]{.custom}` or ```::: {.custom} ... :::``` span/divs get transformed into functions of the same name, `#custom[]`, to be defined in the template. I use it quite a lot in Latex output (wish it was buitl-in, actually, no sure why not?), and now that I'm porting some templates to typst I realise I don't have the same convenient mapping and would have to write lua filters myself. This seems a common enough pattern that it'd be great to have it integrated (perhaps both filters could be merged, and made internal, since the logic would be identical). Ideally quarto would also make a list of such custom spans/divs in the Lua filter, to bypass the need to list them explicitly?

**Comment by mcanouil** [comment]
Related/duplicated discussion:

- https://github.com/quarto-dev/quarto-cli/discussions/9072

**Comment by cscheid** [comment]
Agreed.

If you add the class `block`, Quarto will try to convert the div to a `#block()[]`, and there's even a `typst_function_call` internal function in the Lua filter chain. But Quarto should expose this.

**Reply by baptiste** [reply]
Thanks, sounds great!

**Comment by cderv** [comment]
For anyone interest, @christopherkenny did an extension for this : https://github.com/christopherkenny/typst-function
